### ISSUE TITLE
wrong constant name CFUNC

### ISSUE THREAD

**Issue by petenorth** [oracle/truffleruby#2228]
**truffleruby 21.1.0-dev-70c024bb, like ruby 2.7.2, GraalVM CE Native [x86_64-linux]**

was working with truffleruby 21.1.0-dev-4f309b51, like ruby 2.7.2, GraalVM CE Native [x86_64-linux]

One of our CI jobs has started failing with TruffleRuby

```
The command '/bin/sh -c bundle exec rake assets:precompile RAILS_ENV=production NODE_ENV=production SECRET_KEY_BASE=stubbed SKIP_REDIS=true' returned a non-zero code: 1
rake aborted!
NameError: wrong constant name CFUNC
constants.c:23:in `rb_const_get'
constants.c:23:in `rb_const_get'
<internal:core> core/kernel.rb:234:in `gem_original_require'
<internal:core> core/kernel.rb:234:in `gem_original_require'
/app/config/application.rb:18:in `<top (required)>'
<internal:core> core/kernel.rb:293:in `require_relative'
/app/Rakefile:4:in `<top (required)>'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
/truffleruby-head/bin/bundle:42:in `<main>'
```

**Comment by eregon** [comment]
Thank you from the report.
Is it from https://github.com/petenorth/schools-experience ?

Could you run `rake` with `--backtrace`?
Otherwise `rake` removes all interesting parts from the backtrace.

Maybe it's something we should consider changing in rake by making a PR: hiding the useful parts of the backtrace by default and showing `<internal` entries does not make sense.

**Comment by petenorth** [comment]
He is the output with `--backtrace`

```
Step 3/3 : RUN bundle exec rake --backtrace assets:precompile RAILS_ENV=production NODE_ENV=production SECRET_KEY_BASE=stubbed SKIP_REDIS=true
 ---> Running in 3cbf9c205326
rake aborted!
NameError: wrong constant name CFUNC
constants.c:23:in `rb_const_get'
constants.c:23:in `rb_const_get'
/truffleruby-head/lib/gems/gems/pg-1.2.2/ext/pg_coder.c:32:in `pg_coder_init_encoder'
/truffleruby-head/lib/gems/gems/pg-1.2.2/ext/pg_coder.c:69:in `pg_simple_encoder_allocate'
/truffleruby-head/lib/truffle/truffle/cext.rb:1194:in `__allocate__'
/truffleruby-head/lib/gems/gems/pg-1.2.2/lib/pg/basic_type_mapping.rb:176:in `register_type'
/truffleruby-head/lib/gems/gems/pg-1.2.2/lib/pg/basic_type_mapping.rb:214:in `<module:BasicTypeRegistry>'
/truffleruby-head/lib/gems/gems/pg-1.2.2/lib/pg/basic_type_mapping.rb:26:in `<top (required)>'
<internal:core> core/kernel.rb:234:in `gem_original_require'
/truffleruby-head/lib/gems/gems/pg-1.2.2/lib/pg.rb:68:in `<module:PG>'
/truffleruby-head/lib/gems/gems/pg-1.2.2/lib/pg.rb:37:in `<top (required)>'
<internal:core> core/kernel.rb:234:in `gem_original_require'
/truffleruby-head/lib/mri/bundler/runtime.rb:74:in `block (2 levels) in require'
/truffleruby-head/lib/mri/bundler/runtime.rb:69:in `each'
/truffleruby-head/lib/mri/bundler/runtime.rb:69:in `block in require'
/truffleruby-head/lib/mri/bundler/runtime.rb:58:in `each'
/truffleruby-head/lib/mri/bundler/runtime.rb:58:in `require'
/truffleruby-head/lib/mri/bundler.rb:174:in `require'
/app/config/application.rb:18:in `<top (required)>'
<internal:core> core/kernel.rb:293:in `require_relative'
/app/Rakefile:4:in `<top (required)>'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
/truffleruby-head/lib/gems/gems/rake-12.3.3/lib/rake/rake_module.rb:29:in `load_rakefile'
/truffleruby-head/lib/gems/gems/rake-12.3.3/lib/rake/application.rb:703:in `raw_load_rakefile'
/truffleruby-head/lib/gems/gems/rake-12.3.3/lib/rake/application.rb:104:in `block in load_rakefile'
/truffleruby-head/lib/gems/gems/rake-12.3.3/lib/rake/application.rb:186:in `standard_exception_handling'
/truffleruby-head/lib/gems/gems/rake-12.3.3/lib/rake/application.rb:103:in `load_rakefile'
/truffleruby-head/lib/gems/gems/rake-12.3.3/lib/rake/application.rb:82:in `block in run'
/truffleruby-head/lib/gems/gems/rake-12.3.3/lib/rake/application.rb:186:in `standard_exception_handling'
/truffleruby-head/lib/gems/gems/rake-12.3.3/lib/rake/application.rb:80:in `run'
/truffleruby-head/lib/gems/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
/truffleruby-head/bin/rake:23:in `<top (required)>'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
/truffleruby-head/lib/mri/bundler/cli/exec.rb:63:in `kernel_load'
/truffleruby-head/lib/mri/bundler/cli/exec.rb:28:in `run'
/truffleruby-head/lib/mri/bundler/cli.rb:476:in `exec'
/truffleruby-head/lib/mri/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/truffleruby-head/lib/mri/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/truffleruby-head/lib/mri/bundler/vendor/thor/lib/thor.rb:399:in `dispatch'
/truffleruby-head/lib/mri/bundler/cli.rb:30:in `dispatch'
/truffleruby-head/lib/mri/bundler/vendor/thor/lib/thor/base.rb:476:in `start'
/truffleruby-head/lib/mri/bundler/cli.rb:24:in `start'
/truffleruby-head/lib/gems/gems/bundler-2.1.4/libexec/bundle:46:in `block in <top (required)>'
/truffleruby-head/lib/mri/bundler/friendly_errors.rb:123:in `with_friendly_errors'
/truffleruby-head/lib/gems/gems/bundler-2.1.4/libexec/bundle:34:in `<top (required)>'
<internal:core> core/kernel.rb:400:in `load'
<internal:core> core/kernel.rb:400:in `load'
/truffleruby-head/bin/bundle:42:in `<main>'
The command '/bin/sh -c bundle exec rake --backtrace assets:precompile RAILS_ENV=production NODE_ENV=production SECRET_KEY_BASE=stubbed SKIP_REDIS=true' returned a non-zero code: 1
```

**Comment by eregon** [comment]
@aardvark179 Could you take a look?
Maybe it relates to the changes about constants in f2e074dd503cc8801f1df8789526c67dd082b513.

**Comment by aardvark179** [comment]
Found the problem. We had the logic backwards in one lookup specialisation that didn't get hit in the C tests. I'll create a PR to fix this.

**Comment by petenorth** [comment]
@aardvark179 confirmed that this is fixed in the latest build.

**Comment by eregon** [comment]
(fix in 2ea201853b5bff2fef25847c6fce3a604f34d266)
